### ISSUE TITLE
Anything that has a variable background should force contrast to shift

### ISSUE THREAD

**Issue by btopro** [elmsln/elmsln#2107]
There's a simple element called paper-article that has a pretty awesome capability baked into it called 'contrast' in which it will automatically shift the text color based on the background color. This is something we should have in our elements that allow for color variability -- 

https://github.com/wasc-io/paper-article/blob/master/paper-article.html#L94

**Comment by btopro** [comment]
```
/**
         * Primary Color.
         */
        primaryColor: {
          type: String,
          value: null,
          reflectToAttribute: true,
          observer: '_primaryColorChanged',
        },
        /**
         * Secondary Color.
         */
        secondaryColor: {
          type: String,
          value: null,
          reflectToAttribute: true,
          observer: '_secondaryColorChanged',
        },
        /**
         * Text color.
         */
        textColor: {
          type: String,
          value: null,
          reflectToAttribute: true,
        },
        /**
         * Primary color class.
         */
        primaryColorClass: {
          type: String,
          reflectToAttribute: true,
          computed: '_computeColorClass(primaryColor)',
        },
        /**
         * Secondary color class.
         */
        secondaryColorClass: {
          type: String,
          reflectToAttribute: true,
          computed: '_computeColorClass(secondaryColor)',
        },
        /**
         * Text color class.
         */
        textColorClass: {
          type: String,
          value: null,
          reflectToAttribute: true,
          computed: '_computeColorClass(textColor)',
        },
/**
       * Make class from color value
       */
      _computeColorClass: function(color) {
        if (color.toLowerCase() == '#ffffff') {
          return 'white-text';
        }
        else if (color == '#000000') {
          return 'black-text';
        }
        else if (color != null && color.substring(0, 1) == '#') {
          return this._colorTransform(color, '', '');
        }
      },
/**
       * Notice secondary color changed
       */
      _secondaryColorChanged: function(newValue, oldValue) {
        if (typeof newValue !== typeof undefined && newValue != null) {
          // sees if there's enough contrast and adjusts them accordingly
          this.validateTextColor('textColor', 'secondaryColor');
        }
      },
      /**
       * Validate and modify the text contrast to ensure the correct contrast
       */
      validateTextColor: function(textprop, bgprop) {
        // verify the value is hex value
        if (this[bgprop].includes('#')) {
          let color = '';
          // strip hash from string
          const colorBuffer = this[bgprop].substring(1);
          // convert rrggbb to decimal
          const rgb = parseInt(colorBuffer, 16);
          // extract RGB
          const r = (rgb >> 16) & 0xff;
          const g = (rgb >> 8) & 0xff;
          const b = (rgb >> 0) & 0xff;
          // calculate luma
          const luma = 0.2126 * r + 0.7152 * g + 0.0722 * b; // per ITU-R BT.709
          // if the luma is to low switch to white text
          if (luma < 141) {
            color = '#ffffff';
          }
          else {
            color = '#000000';
          }
          // Set color and background color vars
          this.set(textprop, color);
        }
      },
```

**Comment by btopro** [comment]
video-player currently does this; should be something we add to other stuff as well. Let's just implement this design pattern now knowing how to do it :)
